src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,87 @@
+import { FC } from 'react';
+import classNames from 'classnames';
+import { Person } from '../../types';
+import { PersonLink } from '../PersonLink/PersonLink';
+
+type Props = {
+  people: Person[];
+  selectedPerson: string;
+};
+
+export const PeopleTable: FC<Props> = ({ people, selectedPerson }) => {
+  const isSelected = (person: Person) => person.slug === selectedPerson;
+
+  return (
+    <table
+      data-cy="peopleTable"
+      className="table is-striped is-hoverable is-narrow is-fullwidth"
+    >
+      <thead>
+        <tr>
+          <th>Name</th>
+          <th>Sex</th>
+          <th>Born</th>
+          <th>Died</th>
+          <th>Mother</th>
+          <th>Father</th>
+        </tr>
+      </thead>
+
+      <tbody>
+        {people.map(person => {
+          const {
+            sex,
+            born,
+            died,
+            slug,
+            fatherName,
+            motherName,
+          } = person;
+
+          return (
+            <tr
+              data-cy="person"
+              key={slug}
+              className={classNames(
+                { 'has-background-warning': isSelected(person) },
+              )}
+            >
+              <td>
+                <PersonLink person={person} />
+              </td>
+
+              <td>{sex}</td>
+              <td>{born}</td>
+              <td>{died}</td>
+
+              <td>
+                {people.some(woman => woman.name === motherName)
+                  ? (
+                    <PersonLink
+                      person={
+                        people.find(
+                          woman => woman.name === motherName,
+                        ) || person
+                      }
+                    />
+                  )
+                  : motherName || '-'}
```

Comment:
Now you need to perform extra iterations on every render, instead - prepare array with data after you receive it from server to avoid unnecessary calculations